test/acceptance/application.test.ts

```diff
@@ -19,375 +20,82 @@ describe('Controllers', () => {
-          title: 'flux capacitor'
-        })).body;
-        plutonium = (await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'plutonium'
-        })).body;
-      });
-
-      it('returns all todo items on a todo', async () => {
-        const result = (await client
-          .get(`/todo/${shopping.id}/items`)
-          .send()
-          .expect(200)).body;
-        expect(result.length).to.eql(2);
-        expect(result).to.containEql(flux);
-        expect(result).to.containEql(plutonium);
-      });
-    });
-    describe('replaceTodoItem', () => {
-      let shopping: Todo;
-      let flux: Todo;
-      before(async () => {
-        shopping = (await client.post('/todo').send({
-          title: 'go shopping'
-        })).body;
-        flux = (await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'flux capacitor'
-        })).body;
-      });
-
-      it('successfully replaces a todo item if it exists', async () => {
-        await client
-          .put(`/todo/${shopping.id}/items/${flux.id}`)
-          .send({
-            title: 'samoflange'
-          })
-          .expect(200);
-      });
-
-      it('returns 404 if a todo item does not exist', async () => {
-        await client
-          .put(`/todo/${shopping.id}/items/999999`)
-          .send({
-            title: 'whoops'
-          })
-          .expect(404);
-      });
-    });
-    describe('updateTodoItem', () => {
-      let shopping: Todo;
-      let flux: Todo;
-
-      before(async () => {
-        shopping = (await client.post('/todo').send({
-          title: 'go shopping'
-        })).body;
-        flux = (await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'flux capacitor'
-        })).body;
-      });
-
-      it('successfully updates todo item if it exists', async () => {
-        const path = `/todo/${shopping.id}/items/${flux.id}`;
-        await client
-          .patch(path)
-          .send({
-            isComplete: true
-          })
-          .expect(200);
-        await client
-          .get(path)
-          .send()
-          .expect(
-            200,
-            Object.assign({}, flux, {
-              isComplete: true
-            })
-          );
-      });
-
-      it('returns 404 if a todo item does not exist', async () => {
-        await client
-          .patch(`/todo/${shopping.id}/items/999999`)
-          .send({
-            title: 'whoops'
-          })
-          .expect(404);
-      });
-    });
-    describe('deleteTodoItem', () => {
-      let shopping: Todo;
-      let flux: Todo;
-      before(async () => {
-        shopping = (await client.post('/todo').send({
-          title: 'go shopping'
-        })).body;
-        flux = (await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'flux capacitor'
-        })).body;
-      });
+  it('gets a todo by ID', async () => {
+    const todo = await givenTodo(payload as Todo);
```

Comment:
I find these `as Todo` casts ugly. I am proposing to define `givenTodo` to accept any object containing a subset of Todo properties.

```ts
async function givenTodo(todo: Partial<Todo>) {
 // ...
}
```

Ideally, `givenTodo` should fill in any missing properties that are required by the model.

See http://loopback.io/doc/en/lb4/Testing-your-application.html#use-test-data-builders again.